bloom-housing/bloom
# 2085/adds top level catchAll exception filter

## Issue Overview

This PR addresses #issue
https://github.com/bloom-housing/bloom/issues/2085


## Description
This adds a top level exception filter where exceptions will bubble up to/through

This is to give a console.error when errors do happen so backend errors that happen throughout the various nextjs built in logic and decorators can be easier to track down

we can add more onto here later as this is a starting point. Next steps from here would be to go through and remove try-catch blocks throughout the backend so that this top level error handler can be fleshed out/function more in-line with nextjs best practice.

I rolled logic into here from the entity not found exception filter, but in a second pass we should remove the exception filter specific to it and possibly remove this custom logic all together if its not needed. 

Also the way we have our logger attached is in line with best practice so no changes will be needed there

## How Can This Be Tested/Reviewed?
In order to review this you can run the tests to verify that all tests passed locally

then to see it in action modify one of the backend e2e tests in a way that will error (such as modifying what should be a string into a number or a similar change)

run the test and see that the console.error fires letting you know what the error is and the stack trace for it 

outside of that this shouldn't affect any current logic/behavior so without triggering something we shouldn't see any real difference

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
